common/src/IO/ZipFileSystem.cpp

```diff
@@ -24,41 +24,80 @@
 #include "IO/DiskFileSystem.h"
 #include "IO/IOUtils.h"
 
+#include <miniz/miniz.h>
+
 #include <cassert>
 #include <cstring>
 #include <memory>
-
-#include <wx/log.h>
-#include <wx/mstream.h>
-#include <wx/zipstrm.h>
+#include <string>
 
 namespace TrenchBroom {
     namespace IO {
-        ZipFileSystem::ZipCompressedFile::ZipCompressedFile(std::shared_ptr<wxZipInputStream> stream, std::unique_ptr<wxZipEntry> entry) :
-        m_stream(std::move(stream)),
-        m_entry(std::move(entry)) {}
+        // MinizArchive
 
-        MappedFile::Ptr ZipFileSystem::ZipCompressedFile::doOpen() const {
-            const auto path = Path(m_entry->GetName().ToStdString());
+        class MinizArchive {
+        public:
+            mz_zip_archive archive;
+
+            MinizArchive(const char* begin, const size_t size) {
+                mz_zip_zero_struct(&archive);
+
+                if (mz_zip_reader_init_mem(&archive, begin, size, 0) != MZ_TRUE) {
+                    throw FileSystemException("Error calling mz_zip_reader_init_mem");
+                }
+            }
+
+            virtual ~MinizArchive() {
+                mz_zip_reader_end(&archive);
+            }
+
+            /**
+             * Helper to get the filename of a file in the zip archive
+             */
+            std::string filename(const mz_uint fileIndex) {
+                // nameLen includes space for the null-terminator byte
+                const mz_uint nameLen = mz_zip_reader_get_filename(&archive, fileIndex, nullptr, 0);
+                if (nameLen == 0) {
```

Comment:
I assume that for an empty string, nameLen does NOT include the null terminator?

Reply:
`mz_zip_reader_get_filename` returns 0 if there's an error looking up the file index. Otherwise it returns (1+ the length of the filename excluding the null terminator). So if the filename was `""` somehow, nameLen would be 1 which includes the null terminator (the rest of the function should work as expected in that case though).